src/network/wait-online/wait-online.c

```diff
@@ -152,14 +152,17 @@ static int parse_argv(int argc, char *argv[]) {
                         break;
 
                 case 'o': {
-                        LinkOperationalState s;
+                        LinkOperationalStateRange range;
+
+                        r = parse_operational_state_range(optarg, &range);
+                        if (r < 0)
+                                return log_error_errno(r, "Invalid operational state range '%s'", optarg);
+
+                        if (range.min == _LINK_OPERSTATE_INVALID && range.max == _LINK_OPERSTATE_INVALID)
```

Comment:
This check can go away. parse_operational_state_range() should always return a valid range.

Reply:
Changed `parse_operational_state_range` to fail on empty strings. Checks on empty strings now happen before calling `parse_operational_state_range` where necessary.